src/Configuration.ts

```diff
@@ -64,6 +66,11 @@ const defaultConfiguration: Configuration = {
   batchCreationIntervalInSeconds: 600,
 
   readDirectoryIntervalInSeconds: 30,
+
+  s3Url: undefined,
```

Comment:
No. In the Issue @lautarodragan said: "We could do this by iterating over the keys of defaultConfiguration if it contained every possible configuration key, which is not the case right now since some of them are optional."

This was my attempt at fixing that. I didn't know what default values to assign to them though.

Reply:
I see I think this works then if @lautarodragan is good with it. The one thing I might consider then is to make all properties required, not optional so that we don't accidentally forget or remove properties from the config.

Reply:
+1 to making them mandatory if it doesn't break anything

> No. In the Issue @lautarodragan said: "We could do this by iterating over the keys of defaultConfiguration if it contained every possible configuration key, which is not the case right now since some of them are optional."

This was a quick idea. This approach is fragile because some coder could unknowingly introduce a new configuration option to the interface but provide no default value for it. 

But I'm okay with that for now, as long as we are aware. Maybe we should leave a comment above the interface?

Reply:
Made them mandatory and all integration tests are still passing. I also added the comment 
`
Provide default value in defaultConfiguration for any new configuration options
`
above the interface.